README.md

```diff
@@ -157,6 +166,19 @@ cmake .. -GNinja -DCMAKE_BUILD_TYPE=DEBUG \
   -DCMAKE_CXX_COMPILER=/usr/bin/clang++-14
 ```
 
+In MacOS, if you install llvm clang, you need to explicitly specify to use llvm clang.
```

Comment:
I recommend to only give 1 working solution, instead of multiple possible solutions and let user decide, which is not necessary.

For example, the followings should be always working (in MacOS):

1. Always instruct user to `brew install llvm@15`
2. Use the following to compile:

```
export PATH="/opt/homebrew/opt/llvm/bin:$PATH"
 export CC="/opt/homebrew/opt/llvm/bin/clang"
 export CXX="/opt/homebrew/opt/llvm/bin/clang++"

mkdir cmake-build-debug  # The directory name can be customized
cd cmake-build-debug
cmake .. -GNinja -DCMAKE_BUILD_TYPE=DEBUG
ninja tiflash
```

Reply:
BTW this explicitly introduces LLVM 15, which should help us adopt more CXX 20 features.

Reply:
The maintainer: "We should let the developers use minimum cost to achieve building tiflash, and support Apple clang can achieve that."

Reply:
llvm 14 and apple clang 14 both are enough.

Reply:
`brew install llvm@15` does not seems to be something hard I think. It even doesn't need any compiling. On the other side, enforce the use of LLVM 15 unlocks newer CXX 20 features.

Reply:
agree, but left it until we can use clang-15 in centos7, there are some problems with compiling and glibc...

Reply:
May be we can drive devcontainer https://github.com/pingcap/tiflash/pull/6112. It solves CentOS 7 glibc issues.